velero-plugins/imagestream/shared.go

```diff
@@ -97,25 +97,39 @@ func GetRegistryEnvsForLocation(location string, namespace string) ([]string, er
 	if err != nil {
 		return nil, fmt.Errorf("errors getting registry env vars: %v", err)
 	}
-	return coreV1EnvVarArrToStringArr(envVars, bsl.Namespace), nil
+	return coreV1EnvVarArrToStringArr(envVars, bsl.Namespace)
 }
 
-func coreV1EnvVarArrToStringArr(envVars []corev1.EnvVar, namespace string) []string {
-	var envVarsStr []string
+func coreV1EnvVarArrToStringArr(envVars []corev1.EnvVar, namespace string) ([]string, error) {
+	var envVarsStrings []string
 	for _, envVar := range envVars {
-		envVarsStr = append(envVarsStr, coreV1EnvVarToString(envVar, namespace))
+		envVarStr, err := coreV1EnvVarToString(envVar, namespace)
+		if err != nil {
+			return nil, err
+		}
+		if envVarStr != "" {
+			envVarsStrings = append(envVarsStrings, envVarStr)
+		}
 	}
-	return envVarsStr
+	return envVarsStrings, nil
 }
-func coreV1EnvVarToString(envVar corev1.EnvVar, namespace string) string {
+func coreV1EnvVarToString(envVar corev1.EnvVar, namespace string) (string, error) {
 	if envVar.ValueFrom != nil && envVar.ValueFrom.SecretKeyRef != nil {
 		secretData, err := getSecretKeyRefData(envVar.ValueFrom.SecretKeyRef, namespace)
 		if err != nil {
```

Comment:
The PR fixes the issue, but should we do more?

My fear is: user sees the new error (`secrets \"oadp-<BSL-name>-aws-registry-secret\" not found`) and thinks "what do I do?", and creates a new JIRA

In a real example, I do not know what I do if I hit this error. I need to create a secret with that name, but what should be its content?

Reply:
valid feedback.

If this secret is missing, it is because OADP-Operator never meant to create this secret for you for BSLs defined outside dpa.

You can force trigger of this code by ([per workaround](https://issues.redhat.com//browse/OADP-3630))
```
oc label BackupStorageLocation $BSL app.kubernetes.io/component=bsl` 
```

OADP Operator will then create the required secret for this bsl as well.

Long term fix would be for this plugin to just read secret from bsl and use that secret instead... will probably be followup.

Reply:
But yeah if you think error should be changed.. could change as well.